components/cdc/src/endpoint.rs

```diff
@@ -586,63 +595,62 @@ impl<T: 'static + RaftStoreRouter<RocksEngine>> Endpoint<T> {
 
     fn register_min_ts_event(&self) {
         let timeout = self.timer.delay(self.min_ts_interval);
-        let tso = Compat::new(self.pd_client.get_tso());
+        let tso = self.pd_client.get_tso();
         let scheduler = self.scheduler.clone();
         let raft_router = self.raft_router.clone();
         let regions: Vec<(u64, ObserveID)> = self
             .capture_regions
             .iter()
             .map(|(region_id, delegate)| (*region_id, delegate.id))
             .collect();
-        let fut = tso.join(timeout.map_err(|_| unreachable!())).then(
-            move |tso: pd_client::Result<(TimeStamp, ())>| {
-                // Ignore get tso errors since we will retry every `min_ts_interval`.
-                let (min_ts, _) = tso.unwrap_or((TimeStamp::default(), ()));
-                // TODO: send a message to raftstore would consume too much cpu time,
-                // try to handle it outside raftstore.
-                for (region_id, observe_id) in regions {
-                    let scheduler_clone = scheduler.clone();
-                    if let Err(e) = raft_router.significant_send(
-                        region_id,
-                        SignificantMsg::LeaderCallback(Callback::Read(Box::new(move |resp| {
-                            if !resp.response.get_header().has_error() {
-                                match scheduler_clone.schedule(Task::MinTS { region_id, min_ts }) {
-                                    Ok(_) | Err(ScheduleError::Stopped(_)) => (),
-                                    Err(err) => panic!(
-                                        "failed to schedule min_ts event, min_ts: {}, error: {:?}",
-                                        min_ts, err
-                                    ),
-                                }
+        let fut = async move {
+            let (tso, _) =
+                futures03::future::join(tso, timeout.compat().map_err(|_| unreachable!())).await;
```

Comment:
Not related to this PR, but I'm a bit confused about the logic and naming...
This _joins_ the tso future and a _timeout_. So we always pushing forward `resolved_ts` with a PD timestamp about one second ago? cc @MyonKeminta 
And the `timeout` name is misleading to me. It makes me think it's a timeout for TSO, but it's something like interval...

Reply:
Oh, I didn't notice this before and now I'm confused too....  @5kbpers  PTAL...

Reply:
Me too, redirecting to @overvenus

Reply:
You know what, I guess its purpose is actually timeout, which mean there should be `select` instead of `join`, and after fixing this the latency of CDC resolved ts may be reduced by nearly a second

Reply:
Agree, it should be like `timeout.then(tso)`

Reply:
`then` is not equal to `select`.

Reply:
@5kbpers I got it. The 1s interval is provided by this code. So the correct form should be `timeout.then(tso)`.

Reply:
And still the latency might be reduced by nearly a second after fixing it. @5kbpers Do you have time to try it out?

Reply:
Ah, by the way the `timeout` is better to be renamed to something like `interval`

Reply:
It seems the modified logic is wrong

Reply:
This is a bug and causes extra latency for TiCDC, it should be sleep -> tso -> confirm leadership -> reschedule the `RegisterMinTsEvent`.

Also, we can reuse the tso to save pd.tso RPC.